src/providers/krb5/krb5_child.c

```diff
@@ -3502,6 +3509,9 @@ int main(int argc, const char *argv[])
         DEBUG(SSSDBG_CRIT_FAILURE, "Failed to send reply\n");
     }
 
+    /* Restore the chain id */
+    sss_chain_id_set(kr->old_chain_id);
```

Comment:
What is the reason to bother with `old_chain_id`? (It is zero anyway.)

Wouldn't it be better to include both "krb5_child started.\n" and "krb5_child completed successfully\n"/"krb5_child failed!\n"?

Reply:
The same question applies to others `*_child`.

Reply:
The new version fixes these items, old_chain_id is not needed as you mention.